### DISCUSSION TITLE
Recuperare stato sensore esternamente da Home Assistant

### DISCUSSION THREAD

**Discussion by officialdavide** [indomus/forum#1000] [GENERALE - Discussioni generiche]
Ciao,
presupponiamo il caso che abbia Home Assistant su un Raspberry, con Conbee2 e un sensore collegato (es: sensore porta/finestra).
Conbee utilizza DeConz oppure ZigBee2MQTT (devo ancora valutare).

Posso in quale modo recuperare lo stato del sensore per poi gestirlo esternamente?

Esempio: ogni 10 secondi analizzo lo stato del sensore e memorizzo il suo stato all'interno di un mio database SQL (una sorta di storico degli stati). Ovviamente sarà mia cura realizzare il programma/script che memorizza lo stato, ma come lo posso recuperare? C'è un esempio/guida da cui partire?

Questa mia attività che si sovrappone al funzionamento di Home Assistant sarà di _intralcio_ e/o fonte di problemi?

**Comment by snakuzzo** [comment]
L'aggiornamento di stato dei sensori a batteria solitamente viene inviato solo nle momento di cambio stato...per la restante durata vanno in deep sleep e non possono essere "interrogati" per recuperare lo stato.
Su HA vedrai l'ultimo stato ovviamente.
Non ho ben chiaro il tuo scopo, ma credo sia inutile fare uno script che memorizzi lo stato su un DB visto che già c'è il suo di DB (di base SQLite... in alternativa poitresti utilizzare anche MySQL o altri).
Ad ogni modo se proprio vuoi fare una roba tua custom con un tuo script puoi recuperare lo stato direttamente da HA utilizzando le [API](https://developers.home-assistant.io/docs/api/rest/#actions) o facendo una select direttamente sul DB.
E no...non si sovrappone ad HA

Però ripeto...trovo tutto ciò inutile. Se magari dettagli meglio la tua necessità possiamo provare a ragionarci per trovare la soluzione più semplice e "pulita"

**Reply by officialdavide** [reply]
In realtà raccolgo questi dati per poi usarli in un'altta applicazione su cui sto lavorando ancora.

Comunque è ottimo, grazie mille per la tua risposta!

In genere, in quali casi è meglio optate per DeConz rispetto a ZigBee2MQTT?

**Reply by snakuzzo** [reply]
Non è che ci sono casi in cui utilizzare l'uno o l'altro. La scelta spetta a te.
Io non ho mai usato Deconz...ma posso dirti che non ho mai letto di chi passava da zigbee2mqtt a deconz...è stato sempre il contrario. A livello di duttilità ed evoluzione nel tempo non c'è paragone...vince zigbee2mqtt a mani basse. Dai un'occhiata ai nostro sito che trovi tutti i dettagli.

Ah...tra le diverse soluzioni c'è anche [ZHA](https://www.home-assistant.io/integrations/zha/)
